tests/kernel/build_git_kernel.pm

```diff
@@ -0,0 +1,54 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: build, install and boot a custom, upstream kernel tree
+#          from an arbitrary git tree
+#
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use power_action_utils 'power_action';
+
+sub run {
+    my $self = shift;
+    my $git_tree = get_required_var('KERNEL_GIT_TREE');
+    my $git_branch = get_var('KERNEL_GIT_BRANCH', 'master');
+
+    $self->select_serial_terminal;
+
+    # download, compile and install a kernel tree from git
+    zypper_call('in git-core ncurses-devel gcc flex bison libelf-devel libopenssl-devel');
+    assert_script_run("git clone $git_tree linux");
+    assert_script_run('cd linux');
+    assert_script_run("git checkout $git_branch");
+    assert_script_run('zcat /proc/config.gz > .config');
```

Comment:
Are all three of those necessary? AFAICT not

Reply:
They are - how would you propose to do it differently?
Of course, the git checkout command is only useful if we want a branch other than master, but in case we want master, this is done so quickly we won't notice the time

Reply:
I was referring to the changes to config, so the line 34-36.

Reply:
We have to assume that we are building some development trees here on a machine running a rather old kernel. So we need to use the Olddefconfig. This ensures we have the kernel configuration as close to the running kernel as possible. As we don't build kernels to support other machines, we use localmodconfig to only compile modules we need on this system. This could be omitted, but results in a significant reduction of build time.

Of course we don't strictly need this, we could just build a defconfig, but we want to use a configuration that is as close to the one running one. So I would say yes, we do need exactly this.

Reply:
What I mean is that currently you're completely rewriting the config three times - shouldn't just "localmodconfig" have the exact same result?

Reply:
Maybe we could skip the olddefconfig, I'm not sure here. But localmodconfig may miss a lot of settings that are built in and unrelated to modules, so we definitely need to base on the old one. The olddefconfig does not hurt, and maybe saves us from questions to answer to the kernel build system.